conf/tls/tls_config.go

```diff
@@ -0,0 +1,110 @@
+package tls
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+
+	"github.com/squareup/pranadb/errors"
+
+	"github.com/alecthomas/kong"
+)
+
+// TLSConfig can be used to build a TLS configuration.
+type TLSConfig struct {
+	// nolint: golint
+	CACert     NamedFileContent `help:"PEM-encoded CA certificate."`
```

Comment:
I still don't really understand why we need CACert.

When I configure a golang http server with TLS I only pass in server cert and server key. Example here: https://github.com/cashapp/pranadb/blob/http_api/api/http/server.go#L80

So I don't really understand why we need an extra cert for gRPC

Reply:
That is to enable mTLS (client provides its TLS cert to the server for authentication). For mTLS with strict validation of certificates the client and the server need to trust the CAs used to generate the client and server cert respectively.

The HTTP api server you linked does not support yet mTLS by looking at the configuration. [Here's an example using `net/http` that works through different level of TLS all the way to mTLS](https://venilnoronha.io/a-step-by-step-guide-to-mtls-in-go).

Reply:
I've implemented TLS client auth many times, mainly in Java and the API is surprising to me here. Typically with client auth you would provide a set of client certificates - one for each client that is allowed to connect. Here only one cert is being passed in. Also the property is called "CACert" this is not very descriptive and not obvious what it contains - if the idea is it contains a client cert, then lets call it, for example, "ClientCert" for clarity.